src/TermsOfUseSigner/TermsOfUseSigner.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {EVCUtil} from "ethereum-vault-connector/utils/EVCUtil.sol";
+
+/// By accessing or using Euler's products and services, I agree to the
+/// - [Terms of Use](https://www.euler.finance/terms),
+/// - [Privacy Policy](https://www.euler.finance/privacy-policy), and
+/// - [Risk Disclosures](https://www.euler.finance/risk-disclosures).
+
+/// @title TermsOfUseSigner
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice A contract that allows users to sign the terms of use.
+contract TermsOfUseSigner is EVCUtil {
+    /// @notice Mapping to store timestamp of last signature for each account and terms of use hash
+    mapping(address => mapping(bytes32 => uint256)) public termsOfUseLastSignatureTimestamps;
```

Comment:
This mapping doesn't need to be `public` if we have the `lastTermsOfUseSignatureTimestamp` accessor below.